### ISSUE TITLE
Figure out when override signatures are checked against declarations.

### ISSUE THREAD

**Issue by atomgalaxy** [atomgalaxy/wg21-cust-points#3]
```cpp
virtual void foo(auto x) = 0;

struct X {
  friend void foo(X x, int y) override; // is this allowed?
};

virtual void foo(auto x, int y) = 0;
```

**Comment by atomgalaxy** [comment]
atomicity
 — 
Today at 1:12 PM
no unless dependent
lewissbaker
 — 
Today at 1:12 PM
issue: resolve when override signatures are checked against declarations.
atomicity
 — 
Today at 1:12 PM
I think one of the "investigate this" things we'll get from EWG is whether we want to have the prototypes be a closed set.
posted.
cor3ntin
 — 
Today at 1:16 PM
no
lewissbaker
 — 
Today at 1:16 PM
what if X was a class template?
cor3ntin
 — 
Today at 1:16 PM
you can't do better than ifndr
atomicity
 — 
Today at 1:20 PM
Depends on whether foo's signature is dependent or not.
basically what we want is "early checking when non-dependent" which is the current practice anyway
lewissbaker
 — 
Today at 1:21 PM
compilers may be able to do partial checking in the case where particular parameter types are non-dependent in both the declaration and implementation.
